### PR TITLE
fix bench_value_from_str() for floats with , for decimal

### PR DESCRIPTION
**Pull request by bp0**
Fixes https://github.com/lpereira/hardinfo/issues/497

### CODE REVIEW

**Comment by ocerman** [comment]
This does not seems to work with my locale:
bench2

Seems like `strtod` is using current locale and because "," is correct decimal separator on my system, now even my result does not display correctly.

**Comment by ocerman** [comment]
@bp0 
Maybe it would be better to save benchmark results in locale-indepedent format and then parsing it would be easier.

**Comment by ocerman** [comment]
OK, I found solution: using `g_ascii_strtod` instead of `strtod` works fine.

File: modules/benchmark.c [hunk 1]
```diff
@@ -60,14 +60,17 @@ char *bench_value_to_str(bench_value r) {
...
     if (str) {
-        c = sscanf(str, "%lf; %lf; %d; %d; %255[^\r\n;|]; %255[^\r\n;|]", &r, &e, &t, &v, extra, user_note);
+        /* try to handle floats from locales that use ',' or '.' as decimal sep */
+        c = sscanf(str, "%[-+0-9.,]; %[-+0-9.,]; %d; %d; %255[^\r\n;|]; %255[^\r\n;|]", rstr, estr, &t, &v, extra, user_note);
         if (c >= 3) {
-            ret.result = r;
-            ret.elapsed_time = e;
+            if ((p = strchr(rstr, ','))) { *p = '.'; }
+            if ((p = strchr(estr, ','))) { *p = '.'; }
+            ret.result = strtod(rstr, NULL);
```
Comment (ocerman) [inline modules/benchmark.c]: `g_ascii_strtod` is local-independent and `.` is always decimal separator

File: modules/benchmark.c [hunk 2]
```diff
@@ -60,14 +60,17 @@ char *bench_value_to_str(bench_value r) {
...
+            ret.elapsed_time = strtod(estr, NULL);
```
Comment (ocerman) [inline modules/benchmark.c]: same as above

**Comment by bp0** [comment]
Right on. 

Maybe integer of fraction of seconds maybe better, idk.

**Comment by lpereira** [comment]
Heh. "eurofloat".

**Pull request merged**